lib/react_test_utils.dart

```diff
@@ -0,0 +1,255 @@
+library react.test_utils;
+
+import 'dart:js';
+import 'dart:html';
+import 'package:react/react.dart';
+import "package:react/react_client.dart" show
+    ReactComponentFactory, ReactComponentFactoryProxy, ReactDOMProxy;
+
+
+const missingAddonsMessage = 'React.addons.TestUtils not found. Ensure you\'ve '
+    'included the React addons in your HTML file.'
+    '\n  This:\n<script src="packages/react/react-with-addons.js"></script>'
+    '\n  Not this:\n<script src="packages/react/react.js"></script>';
+
+JsObject _TestUtils = _getNestedJsObject(
+    context, ['React', 'addons', 'TestUtils'], missingAddonsMessage);
+
+JsObject _Simulate = _TestUtils['Simulate'];
+
+_getNestedJsObject(
+    JsObject base, List<String> keys, [String errorIfNotFound='']) {
+  JsObject object = base;
+  for (String key in keys) {
+    if (!object.hasProperty(key)) {
+      throw 'Unable to resolve '
+          '$key in $base${keys.join('.')}}.\n$errorIfNotFound';
+    }
+    object = object[key];
+  }
+  return object;
+}
+
+JsFunction getFactory(component) {
+  if (component is ReactComponentFactoryProxy) {
+    return (component as ReactComponentFactoryProxy).reactComponentFactory;
+  }
+  return null;
+}
+
+typedef bool ComponentTestFunction(JsObject componentInstance);
+
+/// Event simulation interface.
+///
+/// Provides methods for each type of event that can be handled by a React
+/// component.  All methods are used in the same way:
+///
+///   Simulate.{eventName}([JsObject] element, [Map] eventData)
+///
+/// This should include all events documented at:
+/// http://facebook.github.io/react/docs/events.html
+class Simulate {
+
+  static blur(JsObject element, [Map eventData = const{}]) =>
+      _Simulate.callMethod('blur', [element, eventData]);
```

Comment:
These `eventData` methods need to be translated to `JsObject` instances to work.

I started looking into this and fiddling around with the tests, and then got a little carried away... So, I've made a PR into this branch with some changes that should fix this: https://github.com/matthewbalvanz-wf/react-dart/pull/2